### PR TITLE
[FLINK-9004][tests] Implement Jepsen tests to test job availability.

### PR DESCRIPTION
**Pull request by GJL**
## What is the purpose of the change

*Use the Jepsen framework (https://github.com/jepsen-io/jepsen) to implement
tests that verify Flink's HA capabilities under real-world faults, such as
sudden TaskManager/JobManager termination, HDFS NameNode unavailability, network
partitions, etc. The Flink cluster under test is automatically deployed on YARN
(session & job mode) and Mesos.*

## Brief change log

  - *Implement Jepsen tests.*

## Verifying this change

This change added tests and can be verified as follows:

  - *The changes themselves are tests.*
  - *Run Jepsen tests in docker containers.*
  - *Run unit tests with `lein test`*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no** (at least not to Flink))
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** (but it will as soon as test failures appear) / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

cc: @tillrohrmann @cewood

### CODE REVIEW

File: jepsen-flink/docker/nodes [hunk 1]
```diff
@@ -0,0 +1,3 @@
+n1
```
Comment (GJL) [inline jepsen-flink/docker/nodes]: file must be excluded from RAT plugin

File: jepsen-flink/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+This means that changes to the test sources are immediately reflected in the control node container.
+Moreover, this allows you to test locally built Flink distributions by copying the tarball to the
+project's root and passing a URI with the `file://` scheme to the `run-tests.sh` script, e.g.,
+`file:///jepsen/flink-dist.tgz`.
+
+### Checking the output of tests
+
+Consult the `jepsen.log` file for the particular test run in the `store` folder. The final output of every test will be either
+
+    Everything looks good! ヽ('ー`)ノ
```
Comment (yanghua) [inline jepsen-flink/README.md]: @GJL the end of this line is gibberish？

File: jepsen-flink/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+or
+
+    Analysis invalid! (ﾉಥ益ಥ）ﾉ ┻━┻
```
Comment (yanghua) [inline jepsen-flink/README.md]: the end of this line is gibberish？

File: jepsen-flink/README.md (hunk 2 above)
Comment (zentol) [reply jepsen-flink/README.md]: it's a smiley...

File: jepsen-flink/README.md (hunk 3 above)
Comment (GJL) [reply jepsen-flink/README.md]: it's a "flipping table emoji"

Comment (yanghua) [reply jepsen-flink/README.md]: @GJL I really see a Chinese character, it shows `(ﾉಥ益ಥ）ﾉ ┻━┻` to me in the webpage. It contains a `益` which is a Chinese character. Maybe it's Github's prblem?

Comment (GJL) [reply jepsen-flink/README.md]: The Chinese character is the mouth and and the eyebrows of the emoji: https://github.com/jepsen-io/jepsen/blob/dd197bbce0b92c1ab3423709ac6bb0b2ee853365/jepsen/src/jepsen/core.clj#L532

**Comment by zentol** [comment]
As i understand it we don't want this to be part of the source release, as such we need an exclusion in [create_source_release.sh](https://github.com/apache/flink/blob/master/tools/releasing/create_source_release.sh).

**Comment by zentol** [comment]
whoops, sorry I accidentally closed this PR, added the wrong PR ID to a commit :(
I'm so sorry.

**Comment by GJL** [comment]
@zentol No problem, I opened a new one.

**Pull request closed without merging**